### ISSUE TITLE
Glossary files with same names fails to upload

### ISSUE THREAD

**Issue by DavidMarquezF** [MicrosoftTranslator/DocumentTranslationService#8]
If we send multiple glossary files in different paths but with the same name the app crashes. This is due to how the file names for the blob containers are generated:

https://github.com/MicrosoftTranslator/DocumentTranslationService/blob/c6a0aff7a55550f5b6713240d773369db58c757e/DocumentTranslationBusiness.cs#L508-L511

Since the user is not going to see the result of the glossary file maybe it would be better to generate a GUID:

```c#
Guid.NewGuid().ToString()
```

Maybe even maintain the name in case someone wanted to see the files in the containers:
```c#
Guid.NewGuid().ToString() + Path.GetFileName(filename);
```

**Comment by chriswendt1** [comment]
Hi @DavidMarquezF ,
I suspect you are trying to use multilingual glossaries in a multi-language translation run. The app and the service currently do not support that. In other words the glossary works only when
1) you translate to a single language in this run, OR
2) your glossary contains only "Do Not Translate" (DNT) terms that are the same in all target languages. 

Let me know if you have a need for multilingual glossaries. The app could consume a multilingual glossary, or glossaries, and in the client split it up into bilingual glossaries that are compatible with the service. 

Or let me know your scenario for multiple glossaries, and I can address it better.

**Comment by alexhcheng** [comment]
@chriswendt1 : is there an industry format for multilingual glossary?  

Is this one used at all?
https://community.rws.com/product-groups/trados-portfolio/rws-appstore/w/wiki/6039/multilingual-xml-filetype

We can add support to the API natively, but only if we are not inventing our own file format + standard.

**Comment by chriswendt1** [comment]
@alexhcheng The ISO standardized format for glossaries (official term is "Termbase") is the TBX format.
https://www.tbxinfo.net/developer-resources/tbx-elements-reference/

For instance, Microsoft provides its own terminology in this format: https://www.microsoft.com/en-us/language/Terminology, choose a language, and download terminology to see an example of a TBX. 
The Microsoft downloads are all bilingual only, but TBX allows multiple languages per term.

TBX is not exactly human-readable. The tbxinfo.net site has a converter from spreadsheet with column headers to TBX. I have not tried it, but also shouldn't be that hard to write new. I'll look for some TBX editor, maybe open source.

**Comment by chriswendt1** [comment]
AWS translate reads a CSV format like this:
![image](https://user-images.githubusercontent.com/11096941/172445511-a32ce3d3-10cc-4369-a815-004336296f2e.png)
https://aws.amazon.com/blogs/machine-learning/customize-amazon-translate-output-to-meet-your-domain-and-organization-specific-vocabulary/
This is the layout that Custom Translator consumes as well. I forgot if custom translator does .CSV or .XLSX only. 
It wouldn't hurt to adopt Amazon's format, maybe in addition to TBX.

**Comment by chriswendt1** [comment]
@alexhcheng 
Besides AWS, Google Translate also accepts the multilingual CSV file with BCP-47 headers as described above, and TMX.
https://cloud.google.com/translate/docs/advanced/glossary
Google also has the bilingual header-less CSV just like Document Translation. 

I would say this priority: 
1) Multilingual CSV as AWS and Google and Custom Translator have it.
2) TMX as Google and AWS have it.
3) TBX because it is the right thing, standard-wise. Prio 3 because the other big two don't accept it.

**Comment by chriswendt1** [comment]
Fixed in version 0.9.3.2.
Thank you, @DavidMarquezF for reporting it, and the fix suggestion.
ZIP file in the releases folder is updated. 

This fix is not yet an implementation of multilingual glossaries. Now it is only suitable for a single target language and DNT glossaries.
